### PR TITLE
Reduce the looseness of the default search

### PR DESCRIPTION
**Pull request by roborourke**
The default search query was matching any of the search terms with a fuzziness of 1 allowing for a long tail of results. In some cases search results may not be sorted by relevance and the quantity of results becomes a problem. This removes the default EP search query that performs loose matching on any matching search term so now the longer or more specific a search is the fewer results will be returned.

This also introduces dynamic fuzziness when matching against all the provided search terms depending on the length of the search query.

### CODE REVIEW

**Comment by roborourke** [comment]
@rmccue @joehoyle so we could just remove the fuzzy single word matching query when we're sorting by anything other than relevance but I'm not sure that the long tail of search results is particularly valuable for anyone anyway.

**Comment by joehoyle** [comment]
Hmm yeah changing the results set when changing order by seems maybe odd? You'd see a different amount of pagination when you change the order by which I don't think is very expected?

**Comment by joehoyle** [comment]
@roborourke I wonder if we could set up some test data or something for this. I'd like to also get feedback from the PMs etc who have raised this issue to they can verify the results are closer to expectations.

**Comment by roborourke** [comment]
@joehoyle client was happy with this update, @yumito has more info and I can provide relevant GH issues privately in Slack.

They'd also like to see the ability to use advanced search syntax like supporting quoted text so might be worth making that configurable as part of this PR. It just involves changing to using a `simple_query_string` query in ES.

**Comment by roborourke** [comment]
After a bit more discussion I think for the v4 release we'll use this stricter matching as a default but add config to leave it as the default, in addition we can provide a way to use the more advanced `simple_query_string` engine with support for brackets, and, or, custom fuzziness and quoted strings.

Will update the PR shortly.

File: docs/search-configuration.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+- A word preceded by `-` will be excluded from the results
+
+For example the following search would return results with exact matches for "the law", or the words "judge" and "dred" with some degree of fuzziness and excluding the word "marvel".
+
+```
+"the law" OR (judge AND dred) -marvel
+```
+
+## Strict Search
+By default searching will match all the provided search terms with some degree of fuzziness to allow for misspellings and so on.
```
Comment (joehoyle) [inline docs/search-configuration.md:38]: I'm assuming `strict` is true by default, in which case why do we say "By default searching will match all the provided search terms with some degree of fuzziness to allow for misspellings and so on" which sounds like the opposite of strict?

File: docs/search-configuration.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+This means the more specific a search term is the fewer results will be provided. For example
```
Comment (joehoyle) [inline docs/search-configuration.md:40]: I don't see how the above paragraph "means" this conclusion?

File: docs/search-configuration.md (hunk 1 above)
Comment (roborourke) [reply docs/search-configuration.md:38]: Yeah it is. Hmm, maybe we could flip it to “loose”. It’s essentially the difference between “or” and “and” so we could change it to “default_operator” instead

Comment (roborourke) [reply docs/search-configuration.md:38]: I just thought a Boolean was simpler than a text value

Comment (joehoyle) [reply docs/search-configuration.md:38]: ah ok, I read this paragraph describing not-strict (fuzz), but you are saying "By default search is strict (using AND operators), but does allow for fuzzy word matching still" is that right?

Comment (roborourke) [reply docs/search-configuration.md:38]: Yeah that's correct

Comment (roborourke) [reply docs/search-configuration.md]: I updated this to remove the references to fuzzy matching as it's needlessly confusing. I think that's ok as behind the scenes behaviour.

File: docs/search-configuration.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+## Field Boosting
+By default the post title, excerpt, content, author name and taxonomy terms are searched against. Field boosting allows you to modify the importance of those fields with regards to search results. The following configuration will increase the important of the post title, excerpt and a custom meta field above the default value of 1.
+
+```json
+{
+	"extra": {
+		"altis": {
+			"modules": {
+				"search": {
+					"field-boost" {
```
Comment (joehoyle) [inline docs/search-configuration.md:69]: Missing : here

Comment (rmccue) [inline docs/search-configuration.md:69]:
```suggestion
					"field-boost": {
```

**Pull request merged**